sdks/java/core/src/main/java/org/apache/beam/sdk/io/OffsetBasedSource.java

```diff
@@ -315,6 +330,30 @@ public Double getFractionConsumed() {
     }
 
     @Override
+    public long getParallelismConsumed() {
+      return rangeTracker.getSplitPointsProcessed();
+    }
+
+    @Override
+    public long getParallelismRemaining() {
+      if (isDone()) {
+        return 0;
+      }
+      if (!isStarted()) {
+        // Note that even if the current source does not allow splitting, we don't know that
+        // it's non-empty so we return UNKNOWN instead of 1.
+        return BoundedReader.PARALLELISM_UNKNOWN;
+      }
+      if (!getCurrentSource().allowsDynamicSplitting()) {
```

Comment:
In cases like this I'd suggest using "else if" since it makes it very clear these are alternatives. Otherwise you need to recognize that each if has a return in it.